src/main/java/com/notably/model/BlockModel.java

```diff
@@ -0,0 +1,53 @@
+package com.notably.model;
+
+import com.notably.commons.core.path.AbsolutePath;
+import com.notably.model.block.Block;
+import com.notably.model.block.BlockTree;
+import com.notably.model.block.BlockTreeItem;
+
+/**
+ * API of the BlockModel component.
+ */
+public interface BlockModel {
+    /**
+     * Gets the BlockTree currently in use by Notably.
+     */
+    BlockTree getBlockTree();
+
+    /**
+     * Check if currently open block has the specified child block.
+     */
+    boolean hasChild(Block b);
+
+    /**
+     * Adds a block to the specified path, if possible.
+     */
+    void addBlock(AbsolutePath p, Block b);
+
+    /**
+     * Removes the block at the specified path, if possible.
+     */
+    void removeBlock(AbsolutePath p);
+
+    /**
+     * Opens the block at the specified path, if possible.
+     */
+    BlockTreeItem openBlock(AbsolutePath p);
+
+    /**
+     * Sets the currently open block to a new block. (Replaces its block content)
+     */
+    void saveBlock(Block newBlock);
+
+    /**
+     * Gets the absolute path of the currently open block.
+     * TODO: Revise usage and interaction with other components
+     */
+    AbsolutePath getBlockPath();
```

Comment:
Should we call this `getCurrentlyOpenPath()` instead? Considering we need some adjectives to explicitly indicate that this method is going to return the currently open block's path

Reply:
There should be a `Property<AbsolutePath> currentlyOpenPathProperty()` method and a `void setCurrentlyOpenPath(AbsolutePath path)` too!

Reply:
> There should be a `Property<AbsolutePath> currentlyOpenPathProperty()` method 

Makes sense
> and a `void setCurrentlyOpenPath(AbsolutePath path)` too!

`openBlock(AbsolutePath path)` will achieve this result though. Would explicitly creating `void setCurrentlyOpenPath(AbsolutePath path) be more readable by the API user?

Reply:
> `openBlock(AbsolutePath path)` would achieve this result though (after the suggested change below). Would explicitly creating `void setCurrentlyOpenPath(AbsolutePath path) be more readable by the API user?

Oh right! My bad, I missed this. Yes, `openBlock` does achieve that purpose! I have no preference about which naming to choose, up to you